### PR TITLE
Fix optimizer crash

### PR DESCRIPTION
**Pull request by RandySheriffH**
Fix a crash issue when trying to optimize Conv->Add->Relu.
When there is no nodes after Add, the optimizer should skip the case.
Added a UT for that.

### CODE REVIEW

**Comment by hariharans29** [comment]
Does this fix the issue in https://github.com/microsoft/onnxruntime/issues/8024 perhaps ?

**Comment by RandySheriffH** [comment]
> Does this fix the issue in #8024 perhaps ?

**Comment by RandySheriffH** [comment]
> > Does this fix the issue in #8024 perhaps ?

Do that's a separate thing, referred Cuda APIs has several some shaky versions since 7.6. 
Will look into it.

File: onnxruntime/core/optimizer/conv_activation_fusion.cc [hunk 1]
```diff
@@ -121,6 +121,9 @@ Status ConvActivationFusion::ApplyImpl(Graph& graph, bool& modified, int graph_l
         graph_utils::FinalizeNodeFusion(graph, {conv_node, act_node}, fused_conv);
         modified = true;
       } else if (graph_utils::IsSupportedOptypeVersionAndDomain(next_node, "Add", {6, 7, 13, 14})) {
+        if (next_node.GetOutputEdgesCount() == 0) {
```
Comment (pranavsharma) [inline onnxruntime/core/optimizer/conv_activation_fusion.cc:124]: Can you check if there are other places in this file that need a more robust check?

File: onnxruntime/test/optimizer/graph_transform_test.cc [hunk 2]
```diff
@@ -676,6 +676,24 @@ TEST_F(GraphTransformationTests, FuseCudaConvAddRelu) {
   ASSERT_TRUE(op_to_count["Add"] == 0);
   ASSERT_TRUE(op_to_count["Relu"] == 0);
 }
+
```
Comment (RyanUnderhill) [inline onnxruntime/test/optimizer/graph_transform_test.cc:679]: Only suggestion would be a comment explaining what the bug is that this is trying to detect (or is it obvious to someone familiar with the optimizer?)

**Review by RyanUnderhill — APPROVED** [review]

File: onnxruntime/test/optimizer/graph_transform_test.cc (hunk 2 above)
Comment (RandySheriffH) [reply onnxruntime/test/optimizer/graph_transform_test.cc:680]: Done.

File: onnxruntime/core/optimizer/conv_activation_fusion.cc (hunk 1 above)
Comment (RandySheriffH) [reply onnxruntime/core/optimizer/conv_activation_fusion.cc]: Added check for dependents between Conv and Add, plus another UT model to make the graph being left intact is there are more than one downstream nodes of Add.

**Review by RyanUnderhill — APPROVED** [review]

**Review by pranavsharma — APPROVED** [review]

**Pull request merged**